pkg/virt-controller/watch/vmi/vmi.go

```diff
@@ -171,25 +173,30 @@ func (i informalSyncError) RequiresRequeue() bool {
 	return false
 }
 
+type annotationsGenerator interface {
```

Comment:
While I understand this is done in order to simplify vmi_test by stubbing, what do you think about introducing the interface as part of network/pod/annotations instead? The only annotation generator that actually has the `GenerateFromPod` method is the network one, I think that changing the type to `<package>.annotationGenerator` would better provide the necessary context and allow future extensibility.

Reply:
I retract this comment, after discussing it we've concluded that it may serve future generators.

Reply:
This is done in order to decouple the concrete logic from the VMI controller.
As a result of using this interface, we are able to use a stub in the VMI controller's unit tests.

In Go, interfaces are implemented implicitly, which mean we can describe the behavior a struct/function/method expects from its input.
This is the reason the interface in question is present in this specific location.

Exported interfaces should be interfaces which are very common and useful, for example [io.Reader](https://pkg.go.dev/io#Reader).

I assumed this interface could be beneficial to other SIGs, so I haven't prefixed it with `net`.